### ISSUE TITLE
Identify livestreaming and A/V equipment needs

### ISSUE THREAD

**Issue by dcwalk** [ournetworks/2018#26]
**EDIT: 2018-03-10 updated issue to concretely identify the livestreaming equipment and logistic needs**

- who will champion? @benhylau 
- how many people for the Saturday/Sunday? 2 others @darkdrgn2k ??
- what gear do we need?

----
Lets edit this description to build a shortlist of potential livestreaming options and expertise:

- ISOC Canada
- Vanesa, friend of @benhylau with experience recording and works in video content encoding
- hacklab.to contact?
- @Justin417 and @ASoTNetworks have offered to help and may have access to a live streaming rig with 3 cameras and 1 live switcher board

**Comment by dcwalk** [comment]
@patcon helping on that hacklab.to contact

**Comment by patcon** [comment]
If we could build skills in bringing our own data connection, then that could be very cross-useful outside this event (eg. team of people with know-how to bring mobile livestreaming to future protests or actions, h/t g0v)

**Comment by patcon** [comment]
cc: @avwuff (would love to talk in person in case you have any advice -- prelim context: http://ournetworks.ca/)

**Comment by patcon** [comment]
Related to A/V kinda -- these folks sponsored the Ontario Digital Inclusion Summit and had live transcribing happening to the right of stage during whole event and in streams: http://www.ai-live.com/live-captioning-transcription/

Was such great transcription that I thought it was an actual human doing it until the very end

**Comment by dcwalk** [comment]
Hey, I'm down with using this as an event to build capacity, because I want Toronto to have a badass roving A/V crew so we can livestream conf and events in a mutual aid sorta way (I think Chaos and ISOC NY are setting the pace on this and want us to follow!). 

I have load-in and minor A/V experience from previous events but I too often find myself organizing in other ways, and this would likely be the same.

**Comment by patcon** [comment]
My short-list for research would be maybe:
- [Indymedia](https://indymedia.org/)
- [TAO](http://tao.ca)
- [Toronto Media Co-op](http://toronto.mediacoop.ca/) who was active during G20

But seems they're all in sleeping state, except maybe the TAO mailing lists that I can't read.

Having trouble digging up names to track down. Don't you know some TAO people from last year's conf @dcwalk? I don't see a name on the 2017 site.

**Comment by patcon** [comment]
for now, we've got a solution:

@avwuff has Google Fi. It's $60 USD / month for unlimited LTE and works here. [details](https://fi.google.com/about/plan/). He gets up to 10 SIM cards to use with the plan. He has extras on order. He'll loan us one for the day for livestreaming. He'll charge us only for what he doesn't use. (ie. if he uses 2GB at $10/GB, and we use 11GB that does it's part to bring it to $60, then we pay his $40)

So long story short, we can livestream off his sweet hotspot. (They used it to livestream a [rad furry event last week](https://www.twitch.tv/furrykerfluffle/videos/all), and it worked perfectly -- 1.5G for 3h twitch stream)

**Comment by darkdrgn2k** [comment]
Don't think Google Fi is in Toronto though and Cell data in Canada is kinda expensive.

Some quick scans of their websites
Lucymobile.ca 3gb for $25 
Bell - 10gb for $85 + $5/Gb over
Telus- 6gb for $65 + $0.05 /Mb over
Rogers - 5gb for $60  +  $20 per 10GB
Wind/Freedom 12gb for  $75 (but its wind...)

**Comment by darkdrgn2k** [comment]
> Ideally tho we'll be on wifi/wired internet for a main camera? Unless pre-tea brain is missing something so obvious.

Well unlike a two way call, there will be some buffering involved so i doubt some network instability will affect the stream really. a second or two delay people wont care.  The issue with cell is hope the tower is not overloaded and you get a good signal.

But Wifi/Wired would probably be cheaper/faster/better if that can be arranged.

GoogleFi Roaming ... Not to shabby.. With bill protections it looks like about $60 bucks (im assuming usd so ~$77 Canadian)

I'm sure Twitch has some crazy encoders to compress the stream for transport or something. But i area 1.5 gigs is do-able.

Ive also spread the word through my network of ppl to see if anyone has any experience/knowledge/contacts/ideas to do this. But I'm not holding my breath.

**Comment by patcon** [comment]
Cool! thanks for digging in!

Again, wired is nice, but as someone who co-organizes for a weekly event that changes venue every month, and who loves light event process that is resilient to change (eg. location), i do see some value in learning a good set of processes for running livestreams that doesn't involve negotiation with the venue-provided internet :)

(there can be lots of weird bureaucratic/policy issues and RTMP throttling that we've come across, that throws things off. not that we'll hit that, but people do, and imho would be so so nice to learn practices that generalize and that we can share!)

**Comment by benhylau** [comment]
Agree that having your own backhaul even as a backup would be valuable, though LTE coverage indoors at the venue could get tricky.

**Comment by dcwalk** [comment]
Y'all are awesome.

lets start documenting the  _another nth nested project_ Toronto video collective we've formed. Patcon we gotta use your working open docs and if you leave me near this computer for another couple hours I swear I'll have bought a domain name ;)

**Comment by patcon** [comment]
> LTE coverage indoors at the venue could get tricky

My hotspot at least can identify itself as a USB router. Assuming that's not uncommon, a few [5m USB repeater cords](https://www.monoprice.com/product?p_id=6149) might help? (never used one)

**Comment by ASoTNetworks** [comment]
I also have an hotspot that have ethernet port that we can use for stream

**Comment by darkdrgn2k** [comment]
* USB Dongles/Hubs

I find the USB dongles to work fine, usually. But when they don't, i found yet are not very well backed by their manufacturers and the carriers care even less. 

I been using RAVEN XE (now pretty old) and LS300 from as 3.5G/3.75G bridge (pre-LTE) and they work pretty well and they have 2 external diversity antennas. I think the new LTE models are like LX60 or something. [link](https://www.sierrawireless.com/products-and-solutions/routers-gateways/airlink/). 
They essential bridge the Cellular over to a Ethernet port and whatever you plug into it takes on that IP address (so it works with anything/everything).

I have a spare LS300 but may not be fast enough.

TL;DR:  USB sticks are cheap and easy to come by and USUALLY work.  More enterprise grade solutions are more reliable but are pricey!

* Signal Quality

But regardless of that there still is the issue of Over subscription by the Carriers.  It hasn't happen often but i have been places with strong LTE signal but bandwidth is like 1mpbs. The USB dongles (and many of the "hubs") also may have an external antenna hook up but sometimes they don't even work!

Again USUALLY it works great......... sometimes it doesn't

* 10m USB repeater cords 

The cable is essentially a passive hub at the end of a long USB cable.  I found needing to add an actively Powered USB HUB at the other side to power some power hungry usb devices. And even then i have some pen drives that don't work right for some weird reason.  So be weary of those.  

There are USB over CAT5E extenders. No experience with them but they exist
[link](https://www.startech.com/ca/Cards-Adapters/USB-2/USB-Extenders/1-Port-USB-2-over-Cat5-Cat6-Ethernet-Extender-up-to-330ft-100m~USB2001EXT2 )

**Comment by darkdrgn2k** [comment]
>  10m USB repeater cords

The cable is essentially a passive hub at the end of a long USB cable. I found needing to add an actively Powered USB HUB at the other side to power some power hungry usb devices. And even then i have some pen drives that don't work right for some weird reason. So be weary of those.

There are USB over CAT5E extenders. No experience with them but they exist

**Comment by benhylau** [comment]
@ASoTNetworks @darkdrgn2k can we:

1. Identify equipment needs for OurNetworks budget purposes
2. Schedule a demo with the setup?

@Justin417 do you know if you will make it to the conference with equipment?

**Comment by darkdrgn2k** [comment]
I guess the question is what equipment do we have to work with?

In general for an IPFS stream we will need

- [ ] Camera 
- [ ] Audio (if not part of the camera) or some Tie in to the mic/sound system if any?!?!
- [ ] Some sort of way of encoding h.246 and AAC to support HLC
- [ ] IPFS server to create the IPFS hashes
- [ ] Some place to publish the player & the M3U8 file (ipns is still to slow for this part)
- [ ] An internet connection of some sort with enough bandwidth

As an example I have tested
Raspberry pi 3 with attached Camera
USB sound card with a microphone plugged into it
Raspberry Pi 3 onboard H246 Encoder and Software AAC Encoder
Raspberry Pi 3 as the IPFS server
VPS to host the M3U8 file and Player
Home internet connection

I have also used a HDPVR from Happening to pull a COMPONENT HD signal and stereo AUDIO stream as a hardware encoder for both.

**Comment by darkdrgn2k** [comment]
> Schedule a demo with the setup?

Although i have opened 10 browser windows and streams from different gateways without a problem I think a scale would be good.

**Comment by dcwalk** [comment]
Thanks for that prelim list @darkdrgn2k, can you be more specific about cameras, mics, cables, tripods, onsite server etc.? as well, specs for VPS (and possible providers? e.g. DO droplet? your own? space somewhere else?)

**Comment by darkdrgn2k** [comment]
The reason i was not specific about it is i have no idea.

I could be a PI on a Desk with a cheap microphone plugged into it.
Or some expensive state of the art system.

Guess question is what resources do we have to work with and what do we want to acomplish.

Justin> Hey guys, my apologies but I can no longer offer any AV equipment for use

Also:
Internet access at Mozilla can be requested they have a gigabit pipe!

**Comment by ASoTNetworks** [comment]
I can provide both the IPFS and internet streams.

**Comment by benhylau** [comment]
In terms of hardware, excluding server and bandwidth costs. What is needed?

**Video**
* I don't know what is an appropriate HD camera

**Audio**
* I think two wireless hands-free mics (since some talks have two presenters, and having a backup would be nice)

There are a few things to discuss perhaps on a call, some may influence cost:

* Programming management (@dcwalk suggested)
* Automatic closed-captioning (@patcon suggested)
* Ability to switch between camera and slide streams (this one may require additional hardware?)
* Transcoding and publishing to IPFS, putting together m3u8 files (this probably incurs server cost)
* Hosting Internet stream & IPFS stream (this probably incurs server cost)
* Backup data card (we will rely on venue 1G network, this is probably not necessary)

Did I miss anything? @darkdrgn2k @ASoTNetworks @dcwalk

**Comment by benhylau** [comment]
Budget-wise we expect to only require purchase of one HDMI capture card and paying < $10 for Digital Ocean servers, so I have updated the budget sheet to reflect that A/V allocation be capped at $100. This is possible because:

* Our venue partner has professional A/V equipment
* @ASoTNetworks and Justin have donated server resources so we have more than enough even without Digital Ocean
* We managed to find all other equipment among the people involved

**Comment by dcwalk** [comment]
I think this has been done and all the tracking is happening in https://github.com/tomeshnet/ipfs-live-streaming

Closing this in favour of that, but also the few specific questions we have in #87 and #88
